bftengine/src/bftengine/IncomingMsgsStorageImp.hpp

```diff
@@ -68,16 +71,18 @@ class IncomingMsgsStorageImp : public IncomingMsgsStorage {
   std::shared_ptr<MsgHandlersRegistrator> msgHandlers_;
   std::chrono::milliseconds msgWaitTimeout_;
 
+  using MessageWithCallback = std::pair<std::unique_ptr<MessageBase>, Callback>;
+
   // New messages are pushed to ptrProtectedQueue.... ; protected by lock
-  std::queue<std::unique_ptr<MessageBase>>* ptrProtectedQueueForExternalMessages_;
+  std::queue<MessageWithCallback>* ptrProtectedQueueForExternalMessages_;
```

Comment:
I know that you used the existing style, but can we make it a smart_ptr?

Reply:
Oh, it's for swapping ...

Reply:
so `swap()` can be used

Reply:
I was thinking of making them unique_ptr, but I decided to not convolute this PR. As an off-topic, I also wanted to make the promise a non-member, in order to allow multiple start/stops of the same object. Can I make these improvements in a separate PR?